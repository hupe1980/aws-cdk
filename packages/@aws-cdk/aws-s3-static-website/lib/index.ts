export * from './static-website';
export * from './lambda-function-associations';
export * from './website-bucket';
export * from './website-alias-record';
