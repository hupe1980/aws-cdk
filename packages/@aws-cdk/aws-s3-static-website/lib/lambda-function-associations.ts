import {
  CfnDistribution,
  CloudFrontWebDistribution
} from '@aws-cdk/aws-cloudfront';
import { Construct } from '@aws-cdk/core';

export interface Association {
  /**
   * The ARN of the Lambda function
   */
  readonly functionArn: string;

  /**
   * The version of the Lambda function
   */
  readonly functionVersion: string;

  /**
   * Specifies the event type that triggers a Lambda function invocation
   */
  readonly eventType:
    | 'viewer-request'
    | 'origin-request'
    | 'origin-response'
    | 'viewer-response';
}

export interface LambdaFunctionAssociationsProps {
  /**
   * The CloudFront distribution
   */
  readonly distribution: CloudFrontWebDistribution;

  /**
   * The Lambda function associations
   */
  readonly assosiations: Association[];
}

export class LambdaFunctionAssociations extends Construct {
  constructor(
    scope: Construct,
    id: string,
    props: LambdaFunctionAssociationsProps
  ) {
    super(scope, id);

    const cfDist = props.distribution.node.findChild(
      'CFDistribution'
    ) as CfnDistribution;

    const assosiations = props.assosiations.map(assosiation => ({
      EventType: assosiation.eventType,
      LambdaFunctionARN: `${assosiation.functionArn}:${
        assosiation.functionVersion
      }`
    }));

    cfDist.addOverride(
      'Properties.DistributionConfig.DefaultCacheBehavior.LambdaFunctionAssociations',
      assosiations
    );
  }
}
