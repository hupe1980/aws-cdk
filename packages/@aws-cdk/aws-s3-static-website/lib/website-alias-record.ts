import {
  ARecord,
  HostedZone,
  IAliasRecordTarget,
  RecordTarget
} from '@aws-cdk/aws-route53';
import { Construct } from '@aws-cdk/core';

export interface WebsiteAliasRecordProps {
  /** The domain name for the site like 'example.com' */
  readonly domainName: string;

  /** Names for the records. */
  readonly recordNames: string[];

  /** Target for the alias record */
  readonly target: IAliasRecordTarget;
}

export class WebsiteAliasRecord extends Construct {
  constructor(scope: Construct, id: string, props: WebsiteAliasRecordProps) {
    super(scope, id);

    const { domainName, recordNames, target } = props;

    const zone = HostedZone.fromLookup(this, 'Zone', {
      domainName
    });

    recordNames.forEach(
      (recordName, idx): void => {
        new ARecord(this, `WebsiteAliasRecord${idx}`, {
          zone,
          recordName: `${recordName}.`,
          target: RecordTarget.fromAlias(target)
        });
      }
    );
  }
}
