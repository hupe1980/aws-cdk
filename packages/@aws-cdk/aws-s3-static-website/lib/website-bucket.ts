import {
  CfnCloudFrontOriginAccessIdentity,
  S3OriginConfig
} from '@aws-cdk/aws-cloudfront';
import { CanonicalUserPrincipal } from '@aws-cdk/aws-iam';
import { Bucket } from '@aws-cdk/aws-s3';
import { BucketDeployment, Source } from '@aws-cdk/aws-s3-deployment';
import { Construct, RemovalPolicy } from '@aws-cdk/core';
import * as path from 'path';

export interface WebsiteBucketProps {
  /**
   * Name of the bucket
   *
   * @default - Assigned by CloudFormation (recommended).
   */
  readonly bucketName?: string;

  /**
   * Policy to apply when the bucket is removed from this stack.
   *
   * @default - The bucket will be orphaned.
   */
  readonly removalPolicy?: RemovalPolicy;

  /**
   * The source from which to deploy the website
   *
   * @default - Dummy placeholder
   */
  readonly source?: string;

  /**
   * Disable website deployment
   *
   * @default - false
   */
  readonly disableUpload?: boolean;

  /**
   * The index page for the site like 'index.html'
   *
   * @default - index.html
   */
  readonly websiteIndexDocument?: string;

  /**
   *  The error page for the site like 'error.html'
   *
   *  @default - error.html
   */
  readonly websiteErrorDocument?: string;
}

export class WebsiteBucket extends Construct {
  public readonly s3OriginConfig: S3OriginConfig;

  constructor(scope: Construct, id: string, props: WebsiteBucketProps = {}) {
    super(scope, id);

    const {
      bucketName,
      removalPolicy = RemovalPolicy.RETAIN,
      disableUpload = false,
      source,
      websiteIndexDocument,
      websiteErrorDocument
    } = props;

    const bucket = new Bucket(this, 'WebsiteBucket', {
      bucketName,
      removalPolicy,
      websiteIndexDocument: websiteIndexDocument || 'index.html',
      websiteErrorDocument: websiteErrorDocument || 'error.html'
    });

    const originId = new CfnCloudFrontOriginAccessIdentity(
      this,
      'OriginAccessIdentity',
      {
        cloudFrontOriginAccessIdentityConfig: {
          comment: `CloudFront OriginAccessIdentity for ${bucket.bucketName}`
        }
      }
    );

    bucket.grantRead(
      new CanonicalUserPrincipal(originId.attrS3CanonicalUserId)
    );

    if (!disableUpload) {
      const placeHolderSource = path.join(__dirname, '..', 'website');

      new BucketDeployment(this, 'WebsiteDeployment', {
        source: Source.asset(source || placeHolderSource),
        destinationBucket: bucket,
        retainOnDelete: removalPolicy === RemovalPolicy.RETAIN
      });
    }

    this.s3OriginConfig = {
      originAccessIdentityId: originId.ref,
      s3BucketSource: bucket
    };
  }
}
