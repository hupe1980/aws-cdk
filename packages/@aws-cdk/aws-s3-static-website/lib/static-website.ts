import {
  AliasConfiguration,
  CloudFrontWebDistribution,
  CloudFrontWebDistributionProps
} from '@aws-cdk/aws-cloudfront';
import { CloudFrontTarget } from '@aws-cdk/aws-route53-targets';
import { Construct } from '@aws-cdk/core';

import {
  Association,
  LambdaFunctionAssociations
} from './lambda-function-associations';
import { WebsiteAliasRecord } from './website-alias-record';
import { WebsiteBucket, WebsiteBucketProps } from './website-bucket';

export interface AliasProps extends AliasConfiguration {
  /** The domain name for the site like 'example.com' */
  readonly domainName: string;
}

export interface StaticWebsiteProps extends WebsiteBucketProps {
  /**
   * BucketConfiguration is used to configured the S3 website bucket
   *
   * @default - The website bucket is provided with default values
   */
  readonly bucketConfiguration?: WebsiteBucketProps;

  /**
   * AliasConfiguration is used to configured CloudFront to respond to requests on custom domain names.
   *
   * @default - No custom domain names are set up
   */
  readonly aliasConfiguration?: AliasProps;

  /**
   * AWS WAF WebACL to associate with this CloudFront distribution
   *
   * @default - No AWS Web Application Firewall web access control list (web ACL)
   */
  readonly webACLId?: string;
}

export class StaticWebsite extends Construct {
  private readonly distribution: CloudFrontWebDistribution;

  constructor(scope: Construct, id: string, props: StaticWebsiteProps = {}) {
    super(scope, id);

    const { aliasConfiguration, bucketConfiguration, webACLId } = props;

    const websiteBucket = new WebsiteBucket(this, 'WebsiteBucket', {
      ...bucketConfiguration
    });

    const distibutionConfig: CloudFrontWebDistributionProps = {
      webACLId,
      originConfigs: [
        {
          s3OriginSource: {
            ...websiteBucket.s3OriginConfig
          },
          behaviors: [{ isDefaultBehavior: true }]
        }
      ],
      aliasConfiguration
    };

    this.distribution = new CloudFrontWebDistribution(
      this,
      'WebSiteDistribution',
      distibutionConfig
    );

    if (aliasConfiguration) {
      new WebsiteAliasRecord(this, 'WebsiteAliasRecord', {
        domainName: aliasConfiguration.domainName,
        recordNames: aliasConfiguration.names,
        target: new CloudFrontTarget(this.distribution)
      });
    }
  }

  public addLambdaFunctionAssociation(assosiation: Association) {
    return this.addLambdaFunctionAssociations([assosiation]);
  }

  public addLambdaFunctionAssociations(assosiations: Association[]) {
    return new LambdaFunctionAssociations(this, 'LambdaFunctionAssociation', {
      distribution: this.distribution,
      assosiations
    });
  }
}
