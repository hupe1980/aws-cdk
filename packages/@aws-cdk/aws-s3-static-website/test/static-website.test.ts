import { Stack } from '@aws-cdk/core';
import '@cloudcomponents/jest-cdk-snapshot';
import { StaticWebsite } from '../lib';

test('default setup', () => {
  const stack = new Stack();

  new StaticWebsite(stack, 'StaticWebsite');

  expect(stack).toMatchCdkSnapshot();
});

test('lambda@edge', () => {
  const stack = new Stack();

  const staticWebsite = new StaticWebsite(stack, 'StaticWebsite');

  staticWebsite.addLambdaFunctionAssociation({
    functionArn: 'arn:aws:lambda:us-east-1:123456789012:function:my-function',
    functionVersion: '1',
    eventType: 'origin-request'
  });

  expect(stack).toMatchCdkSnapshot();
});
