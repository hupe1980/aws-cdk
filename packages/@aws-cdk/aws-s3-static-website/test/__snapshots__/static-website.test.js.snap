// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default setup 1`] = `
Object {
  "Parameters": Object {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeArtifactHashEF37AD24": Object {
      "Description": "Artifact hash for asset \\"Custom::CDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C/Code\\"",
      "Type": "String",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3Bucket6E5FB2B7": Object {
      "Description": "S3 bucket for asset \\"Custom::CDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C/Code\\"",
      "Type": "String",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3VersionKey426156C0": Object {
      "Description": "S3 key for asset version \\"Custom::CDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C/Code\\"",
      "Type": "String",
    },
    "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetArtifactHash63EBE718": Object {
      "Description": "Artifact hash for asset \\"StaticWebsite/WebsiteBucket/WebsiteDeployment/Asset\\"",
      "Type": "String",
    },
    "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3Bucket03B47293": Object {
      "Description": "S3 bucket for asset \\"StaticWebsite/WebsiteBucket/WebsiteDeployment/Asset\\"",
      "Type": "String",
    },
    "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3VersionKeyA2D139CD": Object {
      "Description": "S3 key for asset version \\"StaticWebsite/WebsiteBucket/WebsiteDeployment/Asset\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3Bucket6E5FB2B7",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3VersionKey426156C0",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3VersionKey426156C0",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.6",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "lambda.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3Bucket03B47293",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3Bucket03B47293",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "StaticWebsiteWebsiteBucket089F8BBF",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "StaticWebsiteWebsiteBucket089F8BBF",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "StaticWebsiteWebSiteDistributionCFDistribution5B02D7B8": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "CacheBehaviors": Array [],
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
            ],
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "ForwardedValues": Object {
              "Cookies": Object {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "StaticWebsiteWebsiteBucket089F8BBF",
                  "RegionalDomainName",
                ],
              },
              "Id": "origin1",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketOriginAccessIdentityD517E39F",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "PriceClass": "PriceClass_100",
          "ViewerCertificate": Object {
            "CloudFrontDefaultCertificate": true,
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "StaticWebsiteWebsiteBucket089F8BBF": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "WebsiteConfiguration": Object {
          "ErrorDocument": "error.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "StaticWebsiteWebsiteBucketOriginAccessIdentityD517E39F": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": Object {
            "Fn::Join": Array [
              "",
              Array [
                "CloudFront OriginAccessIdentity for ",
                Object {
                  "Ref": "StaticWebsiteWebsiteBucket089F8BBF",
                },
              ],
            ],
          },
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "StaticWebsiteWebsiteBucketPolicyD13E39CB": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "StaticWebsiteWebsiteBucket089F8BBF",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "StaticWebsiteWebsiteBucketOriginAccessIdentityD517E39F",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "StaticWebsiteWebsiteBucket089F8BBF",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "StaticWebsiteWebsiteBucket089F8BBF",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "StaticWebsiteWebsiteBucketWebsiteDeploymentCustomResourceA9E467DD": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "StaticWebsiteWebsiteBucket089F8BBF",
        },
        "RetainOnDelete": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketName": Object {
          "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3Bucket03B47293",
        },
        "SourceObjectKey": Object {
          "Fn::Join": Array [
            "",
            Array [
              Object {
                "Fn::Select": Array [
                  0,
                  Object {
                    "Fn::Split": Array [
                      "||",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3VersionKeyA2D139CD",
                      },
                    ],
                  },
                ],
              },
              Object {
                "Fn::Select": Array [
                  1,
                  Object {
                    "Fn::Split": Array [
                      "||",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3VersionKeyA2D139CD",
                      },
                    ],
                  },
                ],
              },
            ],
          ],
        },
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;

exports[`lambda@edge 1`] = `
Object {
  "Parameters": Object {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeArtifactHashEF37AD24": Object {
      "Description": "Artifact hash for asset \\"Custom::CDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C/Code\\"",
      "Type": "String",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3Bucket6E5FB2B7": Object {
      "Description": "S3 bucket for asset \\"Custom::CDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C/Code\\"",
      "Type": "String",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3VersionKey426156C0": Object {
      "Description": "S3 key for asset version \\"Custom::CDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C/Code\\"",
      "Type": "String",
    },
    "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetArtifactHash63EBE718": Object {
      "Description": "Artifact hash for asset \\"StaticWebsite/WebsiteBucket/WebsiteDeployment/Asset\\"",
      "Type": "String",
    },
    "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3Bucket03B47293": Object {
      "Description": "S3 bucket for asset \\"StaticWebsite/WebsiteBucket/WebsiteDeployment/Asset\\"",
      "Type": "String",
    },
    "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3VersionKeyA2D139CD": Object {
      "Description": "S3 key for asset version \\"StaticWebsite/WebsiteBucket/WebsiteDeployment/Asset\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3Bucket6E5FB2B7",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3VersionKey426156C0",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CCodeS3VersionKey426156C0",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.6",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "lambda.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3Bucket03B47293",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3Bucket03B47293",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "StaticWebsiteWebsiteBucket089F8BBF",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "StaticWebsiteWebsiteBucket089F8BBF",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "StaticWebsiteWebSiteDistributionCFDistribution5B02D7B8": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "CacheBehaviors": Array [],
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
            ],
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "ForwardedValues": Object {
              "Cookies": Object {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "LambdaFunctionAssociations": Array [
              Object {
                "EventType": "origin-request",
                "LambdaFunctionARN": "arn:aws:lambda:us-east-1:123456789012:function:my-function:1",
              },
            ],
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "StaticWebsiteWebsiteBucket089F8BBF",
                  "RegionalDomainName",
                ],
              },
              "Id": "origin1",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketOriginAccessIdentityD517E39F",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "PriceClass": "PriceClass_100",
          "ViewerCertificate": Object {
            "CloudFrontDefaultCertificate": true,
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "StaticWebsiteWebsiteBucket089F8BBF": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "WebsiteConfiguration": Object {
          "ErrorDocument": "error.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "StaticWebsiteWebsiteBucketOriginAccessIdentityD517E39F": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": Object {
            "Fn::Join": Array [
              "",
              Array [
                "CloudFront OriginAccessIdentity for ",
                Object {
                  "Ref": "StaticWebsiteWebsiteBucket089F8BBF",
                },
              ],
            ],
          },
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "StaticWebsiteWebsiteBucketPolicyD13E39CB": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "StaticWebsiteWebsiteBucket089F8BBF",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "StaticWebsiteWebsiteBucketOriginAccessIdentityD517E39F",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "StaticWebsiteWebsiteBucket089F8BBF",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "StaticWebsiteWebsiteBucket089F8BBF",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "StaticWebsiteWebsiteBucketWebsiteDeploymentCustomResourceA9E467DD": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "StaticWebsiteWebsiteBucket089F8BBF",
        },
        "RetainOnDelete": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketName": Object {
          "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3Bucket03B47293",
        },
        "SourceObjectKey": Object {
          "Fn::Join": Array [
            "",
            Array [
              Object {
                "Fn::Select": Array [
                  0,
                  Object {
                    "Fn::Split": Array [
                      "||",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3VersionKeyA2D139CD",
                      },
                    ],
                  },
                ],
              },
              Object {
                "Fn::Select": Array [
                  1,
                  Object {
                    "Fn::Split": Array [
                      "||",
                      Object {
                        "Ref": "StaticWebsiteWebsiteBucketWebsiteDeploymentAssetS3VersionKeyA2D139CD",
                      },
                    ],
                  },
                ],
              },
            ],
          ],
        },
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
